tests/integration/conftest.py

```diff
@@ -33,20 +36,56 @@ def traefik_app_name_fixture() -> str:
     return "traefik-k8s"
 
 
+@pytest_asyncio.fixture(scope="module", name="build_charm")
+async def build_charm_fixture(ops_test) -> str:
+    """Builds the charm and injects additional configurations into config.yaml.
+
+    This fixture is designed to simulate a feature that is not yet available in charmcraft that
+    allows for the modification of charm configurations during the build process.
+    Three additional configurations, namely flask_foo_str, flask_foo_int, and flask_foo_bool,
```

Comment:
The additional configuration aren't necessarily prefixed with "flask_" so I'd suggest to remove the prefix and keep "foo_str", "foo_int" and "foo_bool"
I'd also add a variable that will get in conflict with existing flask configuration like "application_root" to ensure the "flask_application_root" config doesn't get overriden

Reply:
We should probably also define a more complex configuration as even if not encouraged, some users may define an array or a dict as configuration parameters and we should make sure this is working.
Even if that's pretty similar to "str", I think we should cover this

Reply:
After we switch to flatted config list, I am using the charm config type to determine user-defined configurations' types. But juju charm config type doesn't support dictionary values. Should we treat all user-defined configurations as JSON string? For example, if user want to set the `foo_str` user has to wrap the `foo_str` value inside double quotes. `juju config flask-k8s foo_str='"bar"'`

Reply:
I don't think we should treat all user-defined configuration as JSON string, but we should consider the scenario where a user will then as JSON string.
Therefore if a user defines a variable `foo_dict` and set it to `juju config flask-k8s foo_dict='{ "foo": "bar", "bar": "foo" }'` or same with `foo_array` using `juju config flask-k8s foo_array='["foo", "bar", "foobar"]'`
We should ensure that the charm is properly working and passing this value to the `FLASK_FOO_DICT` and `FLASK_FOO_ARRAY` envvar without breaking the envvar or the charm